azkaban-common/src/main/java/azkaban/utils/ExternalLink.java

```diff
@@ -46,4 +46,23 @@ public boolean isLinkUrlValid() {
     return this.linkUrlValid;
   }
 
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    ExternalLink link = (ExternalLink) o;
+    return Objects.equals(this.topic, link.topic) &&
+        Objects.equals(this.label, link.label) &&
+        Objects.equals(this.linkUrl, link.linkUrl) &&
```

Comment:
nit: It looks a little bit odd to use `Objects.equals` to test if two strings are equal. Personally prefer this.topic.equals(link.topic)

Reply:
I was trying to avoid the case of `null.equals("someString")`